sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -597,22 +603,30 @@ class SymmetricHashJoinStateManager(
     /**
      * Get all values and indices for the provided key.
      * Should not return null.
+     * Note that we will skip nulls explicitly if config setting for the same is
+     * set to true via STATE_STORE_SKIP_NULLS_FOR_STREAM_STREAM_JOINS.
      */
     def getAll(key: UnsafeRow, numValues: Long): Iterator[KeyWithIndexAndValue] = {
-      val keyWithIndexAndValue = new KeyWithIndexAndValue()
-      var index = 0
       new NextIterator[KeyWithIndexAndValue] {
+        private val keyWithIndexAndValue = new KeyWithIndexAndValue()
+        private var index: Long = 0L
+
+        private def hasMoreValues = index < numValues
         override protected def getNext(): KeyWithIndexAndValue = {
-          if (index >= numValues) {
-            finished = true
-            null
-          } else {
+          while (hasMoreValues) {
```

Comment:
Seems like if the last couple of rows are "null" you can get into an index out of bounds issue.  I think you should check within the while look to see if the index is still i within bounds.

Reply:
@jerrypeng - Not sure I understand the case you are referring to ? So basically for each iteration, we are evaluating 

```
          while (hasMoreValues) {
```

which is essentially calling this function
```
        private def hasMoreValues = index < numValues
```

So, this should check for the out of bounds case, right ? Please let me know if I'm missing something here.

Reply:
Yes I'm not sure I see the case of "out of bound". Could you please elaborate?

Reply:
Oh I just realized its a "def" must have thought it was just a variable. This is not an issue then.

Reply:
No worries. Thx